### PR TITLE
feat(application): add application query exceptions

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add application query exceptions. Need to align the exception thrown logic. 

Pure our online & offline discussion:
1. Should not add any custom exception handle logic at the query script level by default. Exceptions on extreme cases may apply.
2. Should not handle slonik exceptions except for the "not found" error. Throw 500 for all the rest.
3. Catch the slonik NotFount error and return a 404 RequestError with a custom error message at the error-handler middleware.
4. Run query script first then delete script on all the delete apis. 

## Testing

GET
image

DELETE
image

@logto-io/eng

### CODE REVIEW

**Review by darcyYe — APPROVED** [review]
LGTM

File: packages/core/src/queries/application.ts [hunk 1]
```diff
@@ -14,15 +14,34 @@ export const findTotalNumberOfApplications = async () => getTotalRowCount(table)
...
+      code: 'entity.not_exists',
+      name: Applications.tableSingular,
+      status: 404,
+    });
+  }
+};
+
+export const findApplicationById = async (id: string) => {
+  try {
+    return await pool.one<Application>(sql`
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/queries/application.ts:31]: `slonik` will throw a recogonizable error (NotFoundError): https://github.com/gajus/slonik#one, maybe we can handle all this error in a middleware.

**Review by wangsijie — CHANGES_REQUESTED** [review]

File: packages/core/src/queries/application.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/queries/application.ts:31]: the exact point would like to discuss. As for `delete application query` and the `buildUpdateWhere` query methods. It throws the exception at the queries level.  Either one works for me.    If handle the errors at a middleware. not sure we could still get the context i.e. table + where details through the default slonik exception.  Need to check

**Comment by simeng-li** [comment]
updated @wangsijie @gao-sun please review. cc: @darcyYe @IceHe @xiaoyijun  FYI, let's keep aligned on the rest API & DB implementations.

File: packages/core/src/queries/application.ts [hunk 2]
```diff
@@ -49,11 +48,6 @@ export const deleteApplicationById = async (id: string) => {
     where id=${id}
   `);
   if (rowCount < 1) {
-    throw new RequestError({
-      code: 'entity.not_exists_with_id',
-      name: Applications.tableSingular,
-      id,
-      status: 404,
-    });
+    throw new NotFoundError();
```
Comment (gao-sun) [inline packages/core/src/queries/application.ts:51]: per discussion this should be a 500? also do we need wrap the logic into an util? cc @wangsijie

Comment (wangsijie) [reply packages/core/src/queries/application.ts:51]: Throw a general error? A bit different from other not found cases.

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/queries/application.ts [hunk 3]
```diff
@@ -49,11 +48,6 @@ export const deleteApplicationById = async (id: string) => {
     where id=${id}
   `);
   if (rowCount < 1) {
-    throw new RequestError({
-      code: 'entity.not_exists_with_id',
-      name: Applications.tableSingular,
-      id,
-      status: 404,
-    });
+    throw new SlonikError('Resource not found');
```
Comment (gao-sun) [inline packages/core/src/queries/application.ts:51]: will it be better to define a new error class say `SlonikDeletionError` which extends `SlonikError` for accurate reference? we can add an issue and do it later

**Review by gao-sun — APPROVED** [review]
lgtm

File: packages/core/src/queries/application.ts (hunk 3 above)
Comment (simeng-li) [reply packages/core/src/queries/application.ts:51]: LOG-1328

**Pull request merged**